### PR TITLE
chore: Cache processing step for Travis builds

### PR DESCRIPTION
**Pull request by shrutiburman**
This PR:
Solves the issue of clearing cache every-time there's an update to one of the dependencies. 
The step introduced runs `npm audit` and checks for the exit code, only for the case when it is 1, the cache is cleared. 

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli-core/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: src/services/cli-http-client.js [hunk 1]
```diff
@@ -159,7 +159,7 @@ class CliRequestClient {
     const moreInfoMessage = more_info ? `See ${more_info} for more info.` : '';
     const error = {
       message: `Error code ${code || 'N/A'} from Twilio: ${message || 'No message provided'}. ${moreInfoMessage}`,
-      code,
+      code: code.toString().substring(0, 2),
```
Comment (shrutiburman) [inline src/services/cli-http-client.js:162]: Following changes are from main's recent [PR] merge(https://github.com/twilio/twilio-cli-core/pull/142).

File: test/services/cli-http-client.test.js [hunk 2]
```diff
@@ -125,5 +125,17 @@ describe('services', () => {
         });
         expect(response.statusCode).to.equal(200);
       });
+
```
Comment (shrutiburman) [inline test/services/cli-http-client.test.js:128]: Following changes are from main's pull, recent PR merge(#142).

**Review by ravali-rimmalapudi — APPROVED** [review]
Looks Good.

**Review by kridai — APPROVED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -5,6 +5,7 @@ node_js:
   - lts/*
   - '10'
 cache:
+  if: $(npm audit --audit-level=moderate) = 1
   npm: false
```
Comment (onuzbee) [inline .travis.yml:9]: [Q] Would we need to print something here to check in logs that cache has been cleared?

File: .travis.yml [hunk 4]
```diff
@@ -5,6 +5,7 @@ node_js:
...
 before_install:
```
Comment (onuzbee) [inline .travis.yml:10]: [Q] Would we need to print something here that the cache has been cleared in case of error, for sanity in travis logs?

**Review by onuzbee — APPROVED** [review]

File: .travis.yml (hunk 4 above)
Comment (shrutiburman) [reply .travis.yml:10]: Aren't travis logs verbose ? I guess it logs each command that's executed. I did have an echo statement for debugging purposes, I can add that back. Np

Comment (onuzbee) [reply .travis.yml:10]: Makes sense, shouldn't be required! :)

**Review by shamantraghav — APPROVED** [review]

**Pull request merged**